Source/WebCore/platform/KeyboardScrollingAnimator.cpp

```diff
@@ -96,6 +85,94 @@ static FloatSize perpendicularAbsoluteUnitVector(ScrollDirection direction)
     return { };
 }
 
+const std::optional<KeyboardScrollingKey> keyboardScrollingKeyForKeyboardEvent(const KeyboardEvent& event)
+{
+    auto* platformEvent = event.underlyingPlatformEvent();
+    if (!platformEvent)
+        return { };
+
+    // PlatformEvent::Char is a "keypress" event.
+    if (!(platformEvent->type() == PlatformEvent::RawKeyDown || platformEvent->type() == PlatformEvent::Char))
+        return { };
+
+    static constexpr std::pair<ComparableASCIILiteral, KeyboardScrollingKey> mappings[] = {
+        { "Down", KeyboardScrollingKey::DownArrow },
+        { "Left", KeyboardScrollingKey::LeftArrow },
+        { "PageDown", KeyboardScrollingKey::PageDown },
+        { "PageUp", KeyboardScrollingKey::PageUp },
+        { "Right", KeyboardScrollingKey::RightArrow },
+        { "Up", KeyboardScrollingKey::UpArrow },
+    };
+    static constexpr SortedArrayMap map { mappings };
+
+    auto identifier = platformEvent->keyIdentifier();
+    if (auto* result = map.tryGet(identifier))
+        return *result;
+
+    if (platformEvent->text().characterStartingAt(0) == ' ')
+        return KeyboardScrollingKey::Space;
+
+    return { };
+}
+
+const std::optional<ScrollDirection> scrollDirectionForKeyboardEvent(const KeyboardEvent& event)
+{
+    auto key = keyboardScrollingKeyForKeyboardEvent(event);
+    if (!key)
+        return { };
+
+    // FIXME (bug 227459): This logic does not account for writing-mode.
```

Comment:
I can't quite work out why the bodies of these methods are showing up as changes? Did they just get moved around in the file?

Reply:
They used to be lambda expressions in KeyboardScrollingAnimator::keyboardScrollForKeyboardEvent. Separating them into two separate functions introduced a couple of minute changes. Should I try to do something to reduce the size of the diff for this file?